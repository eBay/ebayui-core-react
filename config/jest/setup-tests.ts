import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
