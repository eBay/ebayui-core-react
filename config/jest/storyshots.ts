import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
