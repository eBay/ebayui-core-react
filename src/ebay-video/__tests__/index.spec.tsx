import React from 'react'
import requireContext from 'node-require-context'
import { render, screen } from '@testing-library/react'
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayVideo, EbayVideoSource } from '../index'

describe('<EbayVideo>', () => {
    beforeEach(() => render(<EbayVideo
        thumbnail="https://ir.ebaystatic.com/cr/v/c1/ebayui/video/v1/iphone-thumbnail.jpg"
        width={500}
        height={300}
        a11yPlayText="Play"
        a11yLoadText="Loading"
        errorText="Error loading"
        reportText="Report">
        <EbayVideoSource
            src="https://ir.ebaystatic.com/cr/v/c1/ebayui/video/v1/video.mp4"
        />
    </EbayVideo>))

    it('shows video player with poster', () => {
        const video = document.getElementsByTagName('video')[0]
        expect(video).not.toBeNull()
        expect(video.poster).toBe('https://ir.ebaystatic.com/cr/v/c1/ebayui/video/v1/iphone-thumbnail.jpg')
    })

    it('shows error message', () => {
        expect(screen.getByText('Error loading')).toBeInTheDocument()
    })

    it('shows play button', () => {
        expect(screen.getByLabelText('Play')).toBeInTheDocument()
    })

    it('shows loading spinner', () => {
        expect(screen.getByLabelText('Loading')).toBeInTheDocument()
    })
})

initStoryshots({
    config: ({ configure }) => {
        const req = requireContext('./', false, /\.stories\.tsx$/)
        return configure(req, module)
    }
})
