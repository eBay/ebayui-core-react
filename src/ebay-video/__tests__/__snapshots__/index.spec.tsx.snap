// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-video Captions 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Controlled 1`] = `
<DocumentFragment>
  <button
    class="btn btn--secondary"
  >
    Play
  </button>
   Â 
  <button
    class="btn btn--secondary"
  >
    Mute
  </button>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="margin-top: 1em;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Default 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        poster="https://ir.ebaystatic.com/cr/v/c1/ebayui/video/v1/iphone-thumbnail.jpg"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Fail 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Fail Inside Playlist 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Flexible Container 1`] = `
<DocumentFragment>
  <div
    style="width: 100%;"
  >
    <div
      class="video-player video-player--poster"
      style="width: 600px; height: 400px;"
    >
      <div
        class="video-player__container shaka-video-container"
        data-shaka-player-container=""
        shaka-controls="true"
      >
        <video
          class="shaka-video"
          style="width: 600px; height: 400px;"
        />
        <div
          class="shaka-controls-container"
          shown="true"
        >
          <div
            class="shaka-scrim-container"
          />
          <div
            class="shaka-play-button-container"
          >
            <button
              aria-label="Play"
              class="shaka-play-button shaka-no-propagation"
              icon="play"
            />
          </div>
          <div
            class="shaka-server-side-ad-container"
          />
          <div
            class="shaka-bottom-controls shaka-no-propagation"
          >
            <div
              class="shaka-ad-controls shaka-hidden"
            >
              <div
                class="shaka-ad-position shaka-hidden"
              >
                <span
                  class="shaka-ad-position-span"
                />
              </div>
              <div
                class="shaka-ad-counter"
              >
                <span
                  class="shaka-ad-counter-span"
                />
              </div>
            </div>
            <div
              class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
            />
          </div>
        </div>
        <div
          class="shaka-client-side-ad-container shaka-hidden"
        />
        <div
          class="shaka-spinner-container shaka-hidden"
        >
          <div
            class="shaka-spinner"
            hidden=""
          >
            <svg
              class="shaka-spinner-svg"
              viewBox="0 0 30 30"
            >
              <circle
                class="shaka-spinner-path"
                cx="15"
                cy="15"
                fill="none"
                r="14.5"
                stroke-miterlimit="10"
                stroke-width="1"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="video-player__overlay video-player__overlay--hidden"
      >
        <svg
          aria-hidden="true"
          class="icon icon--attention"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-attention"
          />
        </svg>
        <div
          class="video-player__overlay-text"
        >
          An error has occurred
        </div>
      </div>
      <div
        class="video-player__overlay"
      >
        <span
          aria-label="Loading"
          class="progress-spinner progress-spinner--large"
          role="img"
        >
          <svg
            aria-hidden="true"
            class="icon icon--spinner"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-spinner"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Multiple Videos 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        poster="https://ir.ebaystatic.com/cr/v/c1/ebayui/video/v1/iphone-thumbnail.jpg"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Muted Autoplay 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        autoplay=""
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video No Report Button 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Report Text 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-video Single Video 1`] = `
<DocumentFragment>
  <div
    class="video-player video-player--poster"
    style="width: 600px; height: 400px;"
  >
    <div
      class="video-player__container shaka-video-container"
      data-shaka-player-container=""
      shaka-controls="true"
    >
      <video
        class="shaka-video"
        style="width: 600px; height: 400px;"
      />
      <div
        class="shaka-controls-container"
        shown="true"
      >
        <div
          class="shaka-scrim-container"
        />
        <div
          class="shaka-play-button-container"
        >
          <button
            aria-label="Play"
            class="shaka-play-button shaka-no-propagation"
            icon="play"
          />
        </div>
        <div
          class="shaka-server-side-ad-container"
        />
        <div
          class="shaka-bottom-controls shaka-no-propagation"
        >
          <div
            class="shaka-ad-controls shaka-hidden"
          >
            <div
              class="shaka-ad-position shaka-hidden"
            >
              <span
                class="shaka-ad-position-span"
              />
            </div>
            <div
              class="shaka-ad-counter"
            >
              <span
                class="shaka-ad-counter-span"
              />
            </div>
          </div>
          <div
            class="shaka-controls-button-panel shaka-show-controls-on-mouse-over"
          />
        </div>
      </div>
      <div
        class="shaka-client-side-ad-container shaka-hidden"
      />
      <div
        class="shaka-spinner-container shaka-hidden"
      >
        <div
          class="shaka-spinner"
          hidden=""
        >
          <svg
            class="shaka-spinner-svg"
            viewBox="0 0 30 30"
          >
            <circle
              class="shaka-spinner-path"
              cx="15"
              cy="15"
              fill="none"
              r="14.5"
              stroke-miterlimit="10"
              stroke-width="1"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="video-player__overlay video-player__overlay--hidden"
    >
      <svg
        aria-hidden="true"
        class="icon icon--attention"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-attention"
        />
      </svg>
      <div
        class="video-player__overlay-text"
      >
        An error has occurred
      </div>
    </div>
    <div
      class="video-player__overlay"
    >
      <span
        aria-label="Loading"
        class="progress-spinner progress-spinner--large"
        role="img"
      >
        <svg
          aria-hidden="true"
          class="icon icon--spinner"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-spinner"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
